lib/videojs-playlist-ui.js

```diff
@@ -1,25 +1,188 @@
+    return placeholder;
+  }
+
+  let picture = document.createElement('picture');
+  picture.className = 'vjs-playlist-thumbnail';
+
+  if (typeof thumbnail === 'string') {
+    // simple thumbnails
+    let img = document.createElement('img');
+    img.src = thumbnail;
+    picture.appendChild(img);
+  } else {
+    // responsive thumbnails
+
+    // additional variations of a <picture> are specified as
+    // <source> elements
+    for (let i = 0; i < thumbnail.length - 1; i++) {
+      let variant = thumbnail[i];
+      let source = document.createElement('source');
+      // transfer the properties of each variant onto a <source>
+      for (let prop in variant) {
+        source[prop] = variant[prop];
+      }
+      picture.appendChild(source);
+    }
+
+    // the default version of a <picture> is specified by an <img>
+    let variant = thumbnail[thumbnail.length - 1];
+    let img = document.createElement('img');
+    for (let prop in variant) {
+      img[prop] = variant[prop];
+    }
+    picture.appendChild(img);
+  }
+  return picture;
+};
+
+videojs.PlaylistMenuItem = videojs.Component.extend({
+  init: function(player, options) {
+    if (!options.item) {
+      throw new Error('Cannot construct a PlaylistMenuItem without an item option');
+    }
+    // stub out the element so Component doesn't construct one
+    options.el = true;
+    videojs.Component.call(this, player, options);
+    this.el_ = this.createEl(options.item);
+
+    this.item_ = options.item;
+    this.emitTapEvents();
+
+    this.on(['click', 'tap'], (event) => {
+      const currentIndex = player.playlist().indexOf(this.item_);
+      if (currentIndex >= 0) {
+        player.playlist.currentItem(currentIndex);
+      }
+    });
+  },
+  createEl: function(item) {
+    let li = document.createElement('li');
+    li.className = 'vjs-playlist-item';
+
+    // Thumbnail image
+    li.appendChild(createThumbnail(item.thumbnail));
+
+    // Duration
+    if (item.duration) {
+      let duration = document.createElement('time');
+      let time = videojs.formatTime(item.duration);
+      duration.className = 'vjs-playlist-duration';
+      duration.setAttribute('datetime', 'PT0H0M' + item.duration + 'S');
+      duration.appendChild(document.createTextNode(time));
+      li.appendChild(duration);
+    }
+
+    // Name and description
+    let name = document.createElement('cite');
+    name.className = 'vjs-playlist-name';
+    name.appendChild(document.createTextNode(item.name || this.localize('Untitled Video')));
+    li.appendChild(name);
+
+    if (item.description) {
+      let description = document.createElement('p');
+      description.className = 'vjs-playlist-description';
+      description.appendChild(document.createTextNode(item.description));
+      li.appendChild(description);
+    }
+    return li;
+  }
+});
+
+videojs.PlaylistMenu = videojs.Component.extend({
+  init: function(player, options) {
```

Comment:
if you want, you can the [concise object functions](http://babeljs.io/repl/#?experimental=true&playground=true&evaluate=true&loose=false&spec=false&code=var%20foo%20%3D%20{%0A%20%20foo%28x%2C%20y%29%20{%0A%20%20%20%20return%20x%20%2B%20y%0A%20%20}%0A}%0A%0Aconsole.log%28foo.foo%281%2C%202%29%29%3B):
```js
videojs.Component.extend({
  init (player, options) {
    ...
  },
  createPlaylist_ () {
    ...
  }
});
```